six/two/two.cpp

```diff
@@ -553,9 +551,28 @@ std::string Two::_fetchPythonError()
             if (format_exception != NULL) {
                 PyObject *fmt_exc = PyObject_CallFunctionObjArgs(format_exception, ptype, pvalue, ptraceback, NULL);
                 if (fmt_exc != NULL) {
+                    // Unlikely to happen but PyList_Size might fail, handle error
+                    Py_ssize_t len = PyList_Size(fmt_exc);
+                    if (len == -1) {
+                        // don't fetch the actual error or the caller might think
+                        // it was the root cause, while it's not. Just return
+                        // "unknown error".
+                        PyErr_Clear();
+                        goto done;
```

Comment:
This is debatable, but provided how unlikely this might happen, I'd favor a simple approach instead of figuring out how to communicate that "an error occurred while collecting an error" (even if I believe Windows users wouldn't find anything bizarre in such a message)

Reply:
the approach previously was to return 2 values from this function (so here we could return a struct): one with the exception string from python, one containing whatever error we encountered while grabbing the exception.

We don't necessarily need something as complex, but I think a message that says `an error occurred while collecting an error: here's where we failed to collect the error` is more useful when troubleshooting than returning an `unknown error` :)

Reply:
note: we don't DECREF everything we should DECREF before the `goto`: typically `traceback`, `format_exception` and `fmt_exc`